scm-record/src/types.rs

```diff
@@ -74,71 +74,54 @@ pub enum RecordError {
     Bug(String),
 }
 
-/// The Unix file mode. The special mode `0` indicates that the file did not exist.
+/// The file mode.
 #[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, PartialOrd, Ord)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct FileMode(pub usize);
+pub enum FileMode {
+    /// A Read Write Execute style Unix file mode
+    Rwx(usize),
```

Comment:
nit: What do you think about calling it `Present` instead?

Reply:
Yeah, happy to go with that, I definitely wasn't happy with this name! I also considered `Unix`

Reply:
I like `Unix`

Reply:
`Unix` it is! That's in the latest version